src/store/DialogStore.js

```diff
@@ -10,17 +11,27 @@ export default class DialogStore {
 
     @observable open = false;
 
-    @action.bound setOpen(val) {
+    setOpen = debounce((val) => {
+        this.openDialog(val);
+    }, 300, { leading: true, trailing: false });
+
+    @action.bound openDialog(val) {
         if (this.open !== val) {
             this.open = val;
             if (this.open) { setTimeout(() => this.register(), 100); } else { this.unregister(); }
+
+            if (val === true) { // close active dialog.
+                if (activeDialog) { activeDialog.closeDialog(); }
```

Comment:
do you really need a new function for this (`closeDialog` doesn't need to be an `@action` btw)? Please see if you make the implementation as simple possible. Scrutinise every line.